Code/GraphMol/FilterCatalog/FilterCatalog.h

```diff
@@ -129,7 +129,7 @@ class RDKIT_FILTERCATALOG_EXPORT FilterCatalog : public FCatalog {
 
   FilterCatalog(FilterCatalogParams::FilterCatalogs catalogs)
       : FCatalog(), d_entries() {
-    auto temp_params = static_cast<paramType_t>(catalogs);
+    paramType_t temp_params(catalogs);
     setCatalogParams(&temp_params);
```

Comment:
Taking an address of a temporary is a huge code smell here.  I realize that this is largely historical, but perhaps we should fix the API?

Reply:
```  virtual void setCatalogParams(const paramType &params) {
    // if we already have a paramter object throw an exception
    PRECONDITION(!dp_cParams,
                 "A parameter object already exists on the catalog");
    dp_cParams = new paramType(params);
  }
```

Reply:
I'd be ok with making that change, but not as part of this PR. If we're going to change the API it should be in a separate PR so that it's easy to see what was done.

Here it's worth adding a comment that we know that `setCatalogParams()` copies what is passed in.